src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIteratorExtensions.cs

```diff
@@ -281,8 +228,50 @@ public static bool GetKnownVersion(this Span<byte> span, out string knownVersion
                 }
             }
 
-            knownVersion = null;
+            knownVersion = HttpVersion.Unset;
+            length = 0;
             return false;
         }
+
+        public static string VersionToString(HttpVersion httpVersion)
+        {
+            switch (httpVersion)
+            {
+                case HttpVersion.Http10:
+                    return Http10Version;
+                    break;
+                case HttpVersion.Http11:
+                    return Http11Version;
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(httpVersion), httpVersion, null);
+            }
+        }
+        public static string MethodToString(HttpMethod method)
```

Comment:
If this is called frequently, you might want to create an array of strings, cast the enum value to an int, and then use the int to index into the array.